### PR TITLE
Convert `NaiveDate::{from_yo|from_isoywd}` to return `Result`

### PR DESCRIPTION
**Pull request by pitdicker**
cc @Zomtir. I would like to depend on these methods when working on the `Parsed` type, and you don't seem to have started on them yet.

### CODE REVIEW

**Comment by pitdicker** [comment]
Simplified `from_isoywd` a bit and made that into a commit that can apply to the main branch.

File: src/naive/date/mod.rs [hunk 1]
```diff
@@ -216,55 +216,59 @@ impl NaiveDate {
...
+    pub const fn from_isoywd(year: i32, week: u32, weekday: Weekday) -> Result<NaiveDate, Error> {
         let flags = YearFlags::from_year(year);
         let nweeks = flags.nisoweeks();
         if week == 0 || week > nweeks {
-            return None;
+            if week == 0 || week > 53 {
+                return Err(Error::InvalidArgument);
+            } else {
+                return Err(Error::DoesNotExist);
+            }
```
Comment (djc) [inline src/naive/date/mod.rs:271]: Nit, would suggest the following:

```rust
return Err(match week == 0 || week > 53 {
    true => Error::InvalidArgument,
    false => Error::DoesNotExist,
});
```

**Review by djc — APPROVED** [review]

**Comment by pitdicker** [comment]
I'll keep this PR open a little longer until we have a new merge of main with the 0.5.x branch.

File: src/naive/date/mod.rs (hunk 1 above)
Comment (pitdicker) [reply src/naive/date/mod.rs]: For the next time just say 'Use match'. I should know by now :laughing:.

**Pull request merged**